src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -70,12 +79,24 @@ def get(url, headers)
     request_options.set_as(ResponseBodyType::TEXT)
     configure_timeouts(request_options)
     configure_ssl(request_options)
-    response = SyncHttpClient.get(request_options)
+    response = client_get(request_options)
     ruby_response(response)
   end
 
   private
 
+  def client_get(request_options)
+    SyncHttpClient.get(request_options)
+  rescue Java::ComPuppetlabsHttpClient::HttpClientException => e
+    raise Puppet::Server::HttpClientError.new(e.inspect, e)
```

Comment:
I wonder if it might be better to just pass up `e.message` instead of `e.inspect` here.  For consumers who would try to read `message` on the exception we're raising here and get a payload that looks like the serialized body you get from `inspect` would feel a little strange to me.